datadog/resource_datadog_logs_custom_pipeline.go

```diff
@@ -623,6 +661,27 @@ func buildDatadogUserAgentParser(tfProcessor map[string]interface{}) datadog.Use
 	return ddUserAgentParser
 }
 
+func buildDatadogLookupProcessor(tfProcessor map[string]interface{}) datadog.LookupProcessor {
+	ddLookupProcessor := datadog.LookupProcessor{}
+	if tfSource, exists := tfProcessor["source"].(string); exists {
+		ddLookupProcessor.SetSource(tfSource)
+	}
+	if tfTarget, exists := tfProcessor["target"].(string); exists {
+		ddLookupProcessor.SetTarget(tfTarget)
+	}
+	if tfLookupTable, exists := tfProcessor["lookup_table"].([]interface{}); exists && len(tfLookupTable) > 0 {
+		ddLookupTable := make([]string, len(tfLookupTable))
+		for i, tfLookupLine := range tfLookupTable {
+			ddLookupTable[i] = tfLookupLine.(string)
+		}
+		ddLookupProcessor.LookupTable = ddLookupTable
+	}
+	if tfDefaultLookup, exists := tfProcessor["default_lookup"].(string); exists {
```

Comment:
Based on the schema definition,  `default_lookup`, `is_enabled` and the `name` are optional, shouldn't we check they exist before to cast them?
Also, do we need the exists check on required fields?

Reply:
Apparently we are good here @bkabrda will add details for documentation purpose.

Reply:
I was almost certain that this would fail if the typecasting was there and the value was `nil`, but it doesn't. `tfProcessor["default_lookup"].(string)` correctly works in all possible scenarios (the `default_lookup` key is not present in the map, it's present but nil, it's present and non-nil but of a different type than string - in all these cases, `exists` will be `false`). So everything ok here.